src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -674,25 +674,35 @@ private void mapLicense(List<CompletableFuture<?>> futures, License license, Doc
   }
 
   private void mapFundDistibution(List<CompletableFuture<?>> futures, FundDistribution fundDistribution, Document doc) {
-    Optional.ofNullable(mappings.get(Mapping.Field.FUND_ID))
-      .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> fundDistribution.setFundId((String) o))
-        .exceptionally(Mapper::logException)));
+    if (isValidFundId(futures, fundDistribution, doc)) {
+      Optional.ofNullable(mappings.get(Mapping.Field.FUND_CODE))
+        .ifPresent(field -> futures.add(field.resolve(doc)
+          .thenAccept(o -> fundDistribution.setCode((String) o))
+          .exceptionally(Mapper::logException)));
+
+      Optional.ofNullable(mappings.get(Mapping.Field.FUND_PERCENTAGE))
+        .ifPresent(field -> futures.add(field.resolve(doc)
+          .thenAccept(o -> fundDistribution.withDistributionType(FundDistribution.DistributionType.PERCENTAGE)
+            .setValue((Double) o))
+          .exceptionally(Mapper::logException)));
+
+      Optional.ofNullable(mappings.get(Mapping.Field.ENCUMBERANCE))
+        .ifPresent(field -> futures.add(field.resolve(doc)
+          .thenAccept(o -> fundDistribution.setEncumbrance((String) o))
+          .exceptionally(Mapper::logException)));
+    } else {
+      fundDistribution.withDistributionType(null);
+    }
+  }
 
-    Optional.ofNullable(mappings.get(Mapping.Field.FUND_CODE))
-      .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> fundDistribution.setCode((String) o))
-        .exceptionally(Mapper::logException)));
 
-    Optional.ofNullable(mappings.get(Mapping.Field.FUND_PERCENTAGE))
+  private boolean isValidFundId(List<CompletableFuture<?>> futures, FundDistribution fundDistribution, Document doc) {
+    Optional.ofNullable(mappings.get(Mapping.Field.FUND_ID))
```

Comment:
I think it would be more readable if move this part of mapping out of boolean method to the beginning of `mapFundDistibution(...)`